packages/web/docs/src/app/dynamic-meta-tags.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import { useConfig } from '@theguild/components';
+
+const siteDescription =
+  'Fully Open-source schema registry, analytics and gateway for GraphQL federation and other GraphQL APIs';
+const siteName = 'Hive';
+
+function ensureAbsolute(url: string) {
+  if (url.startsWith('/')) {
+    return `https://the-guild.dev/graphql/hive${url.replace(/\/$/, '')}`;
+  }
+
+  return url;
+}
+
+type NormalizedResult = ReturnType<typeof useConfig>['normalizePagesResult'];
+
+function createBreadcrumb(normalizedResult: NormalizedResult) {
+  const activePaths = normalizedResult.activePath.slice();
+
+  if (activePaths[0].route !== '/') {
+    // Add the home page to all pages except the home page
+    activePaths.unshift({
+      route: '/',
+      title: 'Hive',
+      name: 'index',
+      type: 'page',
+      display: 'hidden',
+      children: [],
+      frontMatter: {},
+    });
+  }
+
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'BreadcrumbList',
+    itemListElement: activePaths.map((path, index) => {
+      return {
+        '@type': 'ListItem',
+        position: index + 1,
+        name: path.route === '/' ? 'Hive' : path.title,
+        item: ensureAbsolute(path.route),
+      };
+    }),
+  };
+}
+
+export function DynamicMetaTags() {
+  const { normalizePagesResult } = useConfig();
+
+  // Get the current page path
+  // Because it shows the full path, from top to bottom,
+  // we need to get the last one to get the current page.
+  const pagePath = normalizePagesResult.activePath[normalizePagesResult.activePath.length - 1];
+
+  const isGatewayDocsPage = pagePath.route.includes('/docs/gateway');
+  const suffix = isGatewayDocsPage ? 'Hive Gateway' : 'Hive';
+  const title = `${pagePath.title} - ${suffix}`;
+
+  const {
+    description = `${siteName}: ${siteDescription}`,
+    canonical,
+    ogImage,
+  } = pagePath.frontMatter;
+
+  const canonicalUrl = ensureAbsolute(canonical ?? pagePath.route);
+
+  return (
+    <>
+      <meta charSet="utf-8" />
+      <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
+      <link rel="canonical" href={canonicalUrl} />
+      <meta content="en" httpEquiv="Content-Language" />
+      <title>{title}</title>
+      <meta name="robots" content="index,follow" />
+      <meta name="description" content={description} />
+      {/* Twitter */}
+      <meta name="twitter:card" content="summary_large_image" />
+      <meta name="twitter:site" content="@TheGuildDev" />
+      <meta name="twitter:creator" content="@TheGuildDev" />
```

Comment:
these already added in metadata API by `getDefaultMetadata`, did you use it? https://github.com/the-guild-org/docs/blob/b3db1f8ded7f0e8c69031fa5fafb9f2c6f19465e/packages/components/src/server/theme-layout.tsx#L181-L185

Reply:
sweet, I'll clean it up